src/xtgeo/interfaces/rms/_rms_base.py

```diff
@@ -0,0 +1,86 @@
+"""Shared base for RMS RegularSurface reader/writer."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from xtgeo.common.log import null_logger
+from xtgeo.interfaces.rms.rmsapi_utils import (
+    RmsApiUtils,
+    _StorageTypeRegularSurface as StorageType,
+)
+
+if TYPE_CHECKING:
+    from xtgeo.interfaces.rms._rmsapi_package import RmsProjectType
+
+logger = null_logger(__name__)
+
+__all__ = ["_BaseRmsDataRW"]
+
+
+class _BaseRmsDataRW:
+    """
+    Common init, input validation etc for read/write RMS RegularSurface, Points, etc.
+    """
+
+    def __init__(
+        self,
+        project: RmsProjectType,
+        name: str,
+        category: str,
+        stype: str,
+        realisation: int = 0,
+    ):
+        self.project = project
+        self.name = name
+        self.category = category
+        self.stype = stype
+        self.realisation = realisation
+
+        # Internal state
+        self._rmsapi_utils: RmsApiUtils | None = None
+        self._stype_enum: StorageType | None = None
+
+    def _init_utils(self, readonly: bool) -> None:
+        """Create RmsApiUtils with the selected readonly mode."""
+        self._rmsapi_utils = RmsApiUtils(self.project, readonly=readonly)
+
+    def _validate_inputs(self) -> StorageType:
```

Comment:
Should these validations occur when the class is initialized? If the initial state is detectable as invalid without side effects, it might be better to fail fast up front, rather than wait for a call to cause it.

This method is also doing two things: validating, _and_ returning a validated/correct stype. That should be two methods, and doesn't that mean that `base_class.stype` might be invalid if this method isn't called?